use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,358 @@
+QAT_VENDOR_AND_ID_VF=$(echo ${QAT_PCI_ID_VF/PCI_ID=} | sed 's/:/ /')
+echo $QAT_VENDOR_AND_ID_VF | sudo tee --append /sys/bus/pci/drivers/vfio-pci/new_id
+
+###Loop through all the virtual functions and bind to the vfio driver
+for f in /sys/bus/pci/devices/0000:$QAT_PCI_BUS_PF_1/virtfn*; do QAT_PCI_BUS_VF=$(basename $(readlink $f)); echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/c6xxvf/unbind; echo $QAT_PCI_BUS_VF | sudo tee --append /sys/bus/pci/drivers/vfio-pci/bind; done
+
+#Verify VF’s created
+ls -la /sys/bus/pci/drivers/vfio-pci
+```
+
+###Check Intel QAT virtual functions are enabled<a name="prepare-8"></a>
+If the following command returns empty then the virtual functions are not properly enabled. This command checks the enumerated device IDs for just the virtual functions. Using the Intel QAT as an example, the physical device ID is 37c8 and virtual function device ID is 37c9. The following command checks if vf's are enabled for any of the currently known Intel QAT device ID's. 
+
+```
+for i in 0442 0443 37c9 19e3; do lspci -d 8086:$i; done
+```
+
+##Prepare Kata Containers<a name="prepare-kata"></a>
+
+###Download Kata Kernel Source<a name="prepare-kata-1"></a>
+This follows the instructions from [https://github.com/kata-containers/packaging/tree/master/kernel](https://github.com/kata-containers/packaging/tree/master/kernel).  The config for the kata-kernel is at [https://github.com/kata-containers/packaging/tree/master/kernel/configs](https://github.com/kata-containers/packaging/tree/master/kernel/configs) In this example, the latest kernel supported by Kata is automatically used.  There are some patches that must be installed as well which the build-kernel.sh script should automatically apply, but if you are using a different kernel version then they may need to be manually applied. Since the Kata Containers kernel has a minimal set of kernel flags set, the config must be updated to set some of the CRYPTO and STACKPROTECTOR flags to enabled. This may change with different kernel versions. The following was tested with kernel v4.19.28-41.
+
+```
+mkdir -p $GOPATH
+cd $GOPATH
+go get -v github.com/kata-containers/packaging
+$GOPATH/src/github.com/kata-containers/packaging/kernel/build-kernel.sh setup
+```
+
+###Build Kata Kernel<a name="prepare-kata-1a"></a>
+
+```
+export LINUX_VER=$(ls -d kata*)
+sed -i 's/EXTRAVERSION =/EXTRAVERSION = .qat.container/' $LINUX_VER/Makefile
+###Change kernel config options to support QAT
+sed -i 's/# CONFIG_PCI_IOV is not set/CONFIG_PCI_IOV=Y/' $LINUX_VER/.config
+sed -i 's/# CONFIG_CRYPTO_SHA1 is not set/CONFIG_CRYPTO_SHA1=Y/' $LINUX_VER/.config
+sed -i 's/# CONFIG_CRYPTO_DH is not set/CONFIG_CRYPTO_DH=Y/' $LINUX_VER/.config
+sed -i 's/# CONFIG_STACKPROTECTOR is not set/CONFIG_STACKPROTECTOR=Y/' $LINUX_VER/.config
+$GOPATH/src/github.com/kata-containers/packaging/kernel/build-kernel.sh build
+```
+
+__Note:__ 
```

Comment:
Please see https://github.com/kata-containers/documentation/blob/master/Documentation-Requirements.md#notes.

Reply:
Thanks. I got the first Note but missed I had a second reference. I should have them all.